conda_build/cli/main_debug.py

```diff
@@ -43,39 +41,42 @@ def parse_args(args):
                          "The top-level recipe can be specified by passing 'TOPLEVEL' here"))
     p.add_argument("-a", "--activate-string-only", action="store_true",
                    help="Output only the string to the used generated activation script.  Use this for creating envs in scripted "
-                   "environments.")
+                        "environments.")
 
     # cut out some args from render that don't make sense here
     #    https://stackoverflow.com/a/32809642/1170370
     p._handle_conflict_resolve(None, [('--output', [_ for _ in p._actions if _.option_strings == ['--output']][0])])
     p._handle_conflict_resolve(None, [('--bootstrap', [_ for _ in p._actions if _.option_strings == ['--bootstrap']][0])])
     p._handle_conflict_resolve(None, [('--old-build-string', [_ for _ in p._actions if
                                                               _.option_strings == ['--old-build-string']][0])])
-    args = p.parse_args(args)
-    return p, args
+
+    return p
 
 
-def execute(args):
-    p, _args = parse_args(args)
+ARG_VALIDATORS = (
+    ('recipe_or_package_file_path', valid.validate_is_conda_pkg_or_recipe_dir),
+)
+
+
+@valid.validate_args(ARG_VALIDATORS, get_parser())
```

Comment:
Btw isn't this design more expensive startup-wise? Calling `get_parser` here means the `ArgumentParser` is generated upon import and not upon function call?

Further, can we rework the `validate_args` signature for a cleaner implementation?
```suggestion
@valid.validate_args(
    get_parser,  # pass in function so we can lazily evaluate
    recipe_or_package_file_path=valid.validate_is_conda_pkg_or_recipe_dir,  # define validators as kwargs
)
```

Another thought, since we use this to validate the `ArgumentParser` parameter should this validator actually be embedded in the argument parser? [Either by defining a custom type or a custom action.](https://stackoverflow.com/questions/37471636/python-argument-parsing-validation-best-practices)

```python
def conda_package(string):
    if os.path.isdir(string) or utils.is_conda_pkg(string):
        return string
    raise argparse.ArgumentTypeError


p.add_argument(
    'recipe_or_package_file_path',
    type=conda_package,
    help=...,
)
```

Reply:
@kenodegard , really like that last thought! Much better to use the API our current argument parser provides rather come up with a custom solution.

Check out my latest revisions. I believe it addresses everything you have brought up here.